x/interchainstaking/keeper/proposal_handler.go

```diff
@@ -239,32 +269,32 @@ func (k *Keeper) HandleUpdateZoneProposal(ctx sdk.Context, p *types.UpdateZonePr
 			k.SetZone(ctx, &zone)
 
 			// generate deposit account
-			portOwner := zone.ChainId + ".deposit"
+			portOwner := zone.ID() + ".deposit"
 			if err := k.registerInterchainAccount(ctx, zone.ConnectionId, portOwner); err != nil {
 				return err
 			}
 
 			// generate withdrawal account
-			portOwner = zone.ChainId + ".withdrawal"
+			portOwner = zone.ID() + ".withdrawal"
 			if err := k.registerInterchainAccount(ctx, zone.ConnectionId, portOwner); err != nil {
 				return err
 			}
 
 			// generate perf account
-			portOwner = zone.ChainId + ".performance"
+			portOwner = zone.ID() + ".performance"
 			if err := k.registerInterchainAccount(ctx, zone.ConnectionId, portOwner); err != nil {
 				return err
 			}
 
 			// generate delegate accounts
-			portOwner = zone.ChainId + ".delegate"
+			portOwner = zone.ID() + ".delegate"
 			if err := k.registerInterchainAccount(ctx, zone.ConnectionId, portOwner); err != nil {
 				return err
 			}
 
 			period := int64(k.GetParam(ctx, types.KeyValidatorSetInterval))
 			query := stakingTypes.QueryValidatorsRequest{}
-			err := k.EmitValSetQuery(ctx, zone.ConnectionId, zone.ChainId, query, sdkmath.NewInt(period))
+			err := k.EmitValSetQuery(ctx, zone.ConnectionId, zone.ID(), query, sdkmath.NewInt(period))
```

Comment:
This does not need to happen for subzones.

Reply:
Given that this is already triggered on 127 and is a periodic ICQ, we only ever need to retrigger this if the `connection` was updated.